app/settings/site/editor.directive.js

```diff
@@ -113,6 +121,12 @@ function (
                     ]).then(function (result) {
                         $scope.saving_config = false;
                         updateSiteHeader();
+                        let newLanguage = result[0].language;
+                        let userLanguage = Session.getSessionDataEntry('language');
+                        if ((userLanguage === undefined || userLanguage === null) && $scope.previousLanguage !== newLanguage) {
```

Comment:
I find the naming scheme of newLanguage, userLanguage, previousLanguage is a little bit obtuse would it be possible to differentiate in the naming between user lang and system lang?